README.rst

```diff
@@ -20,33 +20,85 @@ configurations remain synchronized with the state of all databases in the
 cluster. PgQ is used for buffering mutation records until they are collected by
 the relay and forwarded to their final destination.
 
+Usage
+=====
+
+PGShovel is a tool which replicates mutations (``INSERT``, ``UPDATE`` and ``DELETE`` operations) made to a PostgreSQL data out of the database and into an ordered stream of transaction-aware mutation messages.  These messages can then be consumed by 3rd party systems as needed.  Commonly, the mutation stream is consumed by data warehouse operations, cache updating services, search index services or can even be processed into other new streams.
+
+At Disqus PGShovel is actively being developed to replicate changes from our master Postgres database - which contains the core commenting and other relational data - into our data warehousing system. There is further planned work to use the mutation stream to power a denormalized objects database, offering highly available low-latency access to object data.
+
 Concepts
 ========
 
+PGShovel has a few core concepts which are important to understanding it:
+
 Cluster
 -------
 
-A cluster contains replication sets.
+A Cluster contains a collection of Replication Sets.
 
 Replication Set
 ---------------
 
-Replication sets contain a collection of tables that are replicated together.
+Replication Sets contain configuration defining a collection of tables that are replicated together.
+
+Batch Operations
+----------------
+
+The smallest unit of data in PGShovel.  A Batch Operation is a message communicating one of 4 possible actions:
+
+1. BeginOperation
+2. MutationOperation
+3. CommitOperation
+4. RollbackOperation
+
+Ordered sequences of these operations describe, *with transactional awareness*, the mutations made against the tables contained in a Replication Set.
 
 Relay
 -----
 
-The relay consumes batches of mutations for the replication set and forwards
-those mutation batches to their destination.
+Relays are responsible for extracting queued raw mutations made against PostgreSQL, translating those raw mutation into Batch Operations and then logging (relaying) the Batch Operations to a final destination.  **There is exactly one Relay for each Replication Set**.
```

Comment:
I think this also seems a bit specific for high level documentation (similar to the above, I think this should be documented, but not sure this is the right place.)